### PR TITLE
Refactor appveyor wheel build

### PR DESCRIPTION
**Pull request by matthew-brett**
Refactor on top of Andrew Leech's original PR #3.

This also includes a merge of the `daily` branch, to allow building of current
master.

### CODE REVIEW

**Comment by stefanv** [comment]
This is phenomenally helpful; thanks, Matthew.

We may have to re-enable Python 2.7 for 0.14.x point releases.

Builds are at https://ci.appveyor.com/project/scikit-image/scikit-image-wheels

File: .travis.yml [hunk 1]
```diff
@@ -93,8 +64,15 @@ matrix:
...
+    - if [ "$TRAVIS_BRANCH" == "master" ]; then
+          CONTAINER="pre-release";
+          BUILD_COMMIT=${DAILY_COMMIT:-$BUILD_COMMIT};
+      else
+          CONTAINER=wheels;
+          UPLOAD_ARGS="--no-update-index";
+      fi
     - BUILD_DEPENDS="$NP_BUILD_DEP Cython"
-    - TEST_DEPENDS="$NP_TEST_DEP pytest $GEN_DEPS"
+    - TEST_DEPENDS="$NP_TEST_DEP nose pytest $GEN_DEPS"
```
Comment (soupault) [inline .travis.yml]: Why `nose`?

**Comment by soupault** [comment]
@matthew-brett does this look correct - https://github.com/scikit-image/scikit-image-wheels/tree/v0.14.x ?

File: .travis.yml (hunk 1 above)
Comment (soupault) [reply .travis.yml]: Yes, please. I'm very curous.

**Pull request merged**